code/modules/SCPmodular/scps/euclid/SCP-294.dm

```diff
@@ -0,0 +1,71 @@
+GLOBAL_LIST_EMPTY(scp294_reagents)
+
+/obj/machinery/scp294
+	name = "SCP-294"
+	desc = "A standard coffee vending machine."
+	icon = 'code/modules/SCPmodular/spcicon/scpobj/scp294.dmi'
+	icon_state = "coffee_294"
+	layer = 2.9
+	anchored = 1
+	density = 1
+	var/uses_left = 12
+	var/last_use = 0
+	var/restocking_timer = 0
+
+/obj/machinery/scp294/New()
+	..()
+
+	if(!GLOB.scp294_reagents.len)
+		//Chemical Reagents - Initialises all /datum/reagent into a list indexed by reagent id
+		var/paths = subtypesof(/datum/reagent) - /datum/reagent/medicine/adminordrazine
+		for(var/path in paths)
+			var/datum/reagent/D = new path
+			GLOB.scp294_reagents[D.name] = D
+
+/obj/machinery/scp294/examine(mob/user)
+	user << "<b><span class = 'euclid'><big>SCP-294</big></span></b> - [desc]"
+
+/obj/machinery/scp294/attack_hand(mob/user)
+
+	if((last_use + 3 SECONDS) > world.time)
+		visible_message("<span class='notice'>[src] displays NOT READY message.</span>")
+		return
+
+	last_use = world.time
+	if(uses_left < 1)
+		visible_message("<span class='notice'>[src] displays RESTOCKING, PLEASE WAIT message.</span>")
+		return
+
+	var/product = null
+	var/input_reagent = (input("Enter the name of any liquid", "What would you like to drink?") as text)
+		product = find_reagent(input_reagent)
```

Comment:
Why is there a random ass tab in the middle of nowhere near product?

Reply:
I know what the issue here is too TM